test/ShallowWrapper-spec.jsx

```diff
@@ -2683,7 +2683,131 @@ describe('shallow', () => {
     });
   });
 
+  describe('disableLifecycleMethods', () => {
+    describe('validation', () => {
+      it('throws for a non-boolean value', () => {
+        ['value', 42, null].forEach((value) => {
+          expect(() => shallow(<div />, {
+            disableLifecycleMethods: value,
+          })).to.throw(/true or false/);
+        });
+      });
+
+      it('does not throw for a boolean value or undefined', () => {
+        [true, false, undefined].forEach((value) => {
+          expect(() => shallow(<div />, {
+            disableLifecycleMethods: value,
+          })).not.to.throw();
+        });
+      });
+
+      it('does not throw when no lifecycle flags are provided in options', () => {
+        expect(() => shallow(<div />, {})).not.to.throw();
+      });
+
+      it('throws when used with lifecycleExperimental in invalid combinations', () => {
+        [true, false].forEach((value) => {
+          expect(() => shallow(<div />, {
+            lifecycleExperimental: value,
+            disableLifecycleMethods: value,
+          })).to.throw(/same value/);
+        });
+      });
+    });
+
+    describe('when set to true', () => {
+      let wrapper;
+      const spy = sinon.spy();
+      class Foo extends React.Component {
+        componentWillMount() {
+          spy('componentWillMount');
+        }
+        componentDidMount() {
+          spy('componentDidMount');
+        }
+        componentWillReceiveProps() {
+          spy('componentWillReceiveProps');
+        }
+        shouldComponentUpdate() {
+          spy('shouldComponentUpdate');
+          return true;
+        }
+        componentWillUpdate() {
+          spy('componentWillUpdate');
+        }
+        componentDidUpdate() {
+          spy('componentDidUpdate');
+        }
+        componentWillUnmount() {
+          spy('componentWillUnmount');
+        }
+        render() {
+          spy('render');
+          return <div>foo</div>;
+        }
+      }
+
+      beforeEach(() => {
+        wrapper = shallow(<Foo />, { disableLifecycleMethods: true });
+        spy.reset();
+      });
+
+      it('does not call componentDidMount when mounting', () => {
+        wrapper = shallow(<Foo />, { disableLifecycleMethods: true });
+        expect(spy.args).to.deep.equal([
+          ['componentWillMount'],
+          ['render'],
+        ]);
+      });
+
+      it('calls expected methods when receiving new props', () => {
+        wrapper.setProps({ foo: 'foo' });
+        expect(spy.args).to.deep.equal([
+          ['componentWillReceiveProps'],
+          ['shouldComponentUpdate'],
+          ['componentWillUpdate'],
+          ['render'],
```

Comment:
I'm not sure whether `componentDidUpdate` is supposed to be called here, but it's not currently. If this behavior is correct it's going to make this flag + documentation pretty confusing since, per next test down, setState *does* trigger `componentDidUpdate` for shallow rendered components (even without enzyme in the picture).

The flag is looking more like it should be disableDidMountAllTheTimeAndDidUpdateSomeOfTheTime. 😅 

cc @koba04 any thoughts on correctness here? Note this is all without lifecycleExperimental set to true.